pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -307,6 +311,23 @@ func (vc *VirtualCenter) connect(ctx context.Context, requestNewSession bool) er
 	}
 	// If session has expired, create a new instance.
 	log.Warnf("Creating a new client session as the existing one isn't valid or not authenticated")
+	if vc.reloadConfigForNewClient {
+		cfg, err := config.GetConfig(ctx)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to read config. Error: %+v", err)
+		}
+		newVcenterConfigs, err := GetVirtualCenterConfigs(ctx, cfg)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to get VirtualCenterConfigs. err=%v", err)
+		}
+		for _, newvcconfig := range newVcenterConfigs {
+			if newvcconfig.Host == vc.Config.Host {
+				vc.Config = newvcconfig
+				vc.reloadConfigForNewClient = true
```

Comment:
This bool is set to true only here. But, this code will execute only when `reloadConfigForNewClient` is true as per Line : 314. 
Are we missing anything ?

Reply:
We are setting `vcenterconfig.ReloadVCConfigForNewClient = true` in the CSI controller init when multi vCenter feature is enabled.

Reply:
But ReloadVCConfigForNewClient and reloadConfigForNewClient are 2 separate variables right? It doesn't look like we are initialising `reloadConfigForNewClient` that is being used i this method.

Reply:
Done. Kept only `vc.Config.ReloadVCConfigForNewClient `